config/src/test/java/org/hyperledger/besu/config/CliqueConfigOptionsTest.java

```diff
@@ -64,6 +65,20 @@ public void shouldGetDefaultBlockPeriodFromDefaultConfig() {
         .isEqualTo(EXPECTED_DEFAULT_BLOCK_PERIOD);
   }
 
+  @Test
+  public void shouldThrowOnDecimalBlockPeriod() {
+    final CliqueConfigOptions config = fromConfigOptions(singletonMap("BlockPeriodSeconds", 1.5));
+    assertThatThrownBy(() -> config.getBlockPeriodSeconds())
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void shouldThrowOnNonPositiveBlockPeriod() {
+    final CliqueConfigOptions config = fromConfigOptions(singletonMap("BlockPeriodSeconds", -1));
+    assertThatThrownBy(() -> config.getBlockPeriodSeconds())
+        .isInstanceOf(IllegalArgumentException.class);
+  }
```

Comment:
I think these would better live in PositiveNumberTest.
But then you might still have one in the *ConfigOptionsTest classes that's something like "shouldThrowForNonPositiveInteger"